test/any_object_test.cpp

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unifex/any_object.hpp>
+
+#include <unifex/overload.hpp>
+#include <unifex/tag_invoke.hpp>
+#include <unifex/this.hpp>
+#include <unifex/type_index.hpp>
+
+#include <gtest/gtest.h>
+
+#include <unifex/detail/prologue.hpp>
+
+static_assert(unifex::is_callable_v<unifex::detail::_destroy_cpo, int&>);
+static_assert(unifex::detail::supports_type_erased_cpo_v<
+              int,
+              unifex::detail::_destroy_cpo>);
+
+namespace
+{
+  inline constexpr struct get_typeid_cpo {
+    using type_erased_signature_t =
+        unifex::type_index(const unifex::this_&) noexcept;
+
+    template <typename T>
+    friend unifex::type_index tag_invoke(get_typeid_cpo, const T& x) {
+      return unifex::type_id<T>();
+    }
+
+    template(typename T)                                     //
+        (requires unifex::tag_invocable<get_typeid_cpo, T>)  //
+        auto
+        operator()(const T& x) const noexcept
+        -> unifex::tag_invoke_result_t<get_typeid_cpo, const T&> {
+      static_assert(std::is_same_v<
+                    unifex::type_index,
+                    unifex::tag_invoke_result_t<get_typeid_cpo, const T&>>);
+      return unifex::tag_invoke(get_typeid_cpo{}, x);
+    }
+  } get_typeid{};
```

Comment:
This looks familiar; is there already a similar thing in another test?  Might be worth having a "test utils" directory for stuff like this.

Reply:
I was wondering if we should just add this as a first-class CPO to the library itself.
I can see potential utility for it - e.g. to provide something similar to the `std::any::type()` method to get the `std::type_info` of the current object.